Week3/homework/step2-2.js

```diff
@@ -3,18 +3,26 @@
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const numbers = [];
 
-  // Replace this comment and the next line with your code
-  console.log(startIndex, stopIndex, threeCallback, fiveCallback, numbers);
+  for (let i = startIndex; i <= stopIndex; i++) {
+    numbers.push(i);
+  }
+
+  numbers.forEach(number => {
+    if (number % 3 === 0) {
+      threeCallback(number);
+    }
+    if (number % 5 === 0) {
+      fiveCallback(number);
+    }
+  });
```

Comment:
Couple things here:
1. You can combine the two loop iterations for creating the numbers array and the `if` conditions.
2. I think you missed the requirement where you were also supposed to log on console if a number was divisible by both 3 and 5. Hint: a number which is divisible by both 3 and 5 will also be divisible by 15. 🙂

Reply:
function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
  const numbers = [];

  for (let i = startIndex; i <= stopIndex; i++) {
    numbers.push(i);
  }

  numbers.forEach(number => {
    if (number % 15 === 0) {
      console.log(`${number}, 'can be divided by 3, 5 and 15'`);
    } else if (number % 3 === 0) {
      threeCallback(number);
    } else if (number % 5 === 0) {
      fiveCallback(number);
    }
  });
}

function sayThree(number) {
  console.log(`${number}, 'can be divided by 3'`);
}

function sayFive(number) {
  console.log(`${number}, 'can be divided by 5'`);
}

threeFive(10, 15, sayThree, sayFive);

Reply:
i wrote something like that i works , but I am not sure if it's eneoug for the homework requirement